tests/BatchedProcessorTest.php

```diff
@@ -100,7 +138,60 @@ public function testBatching() {
 		$this->assertEquals(9, $data->currentStep);
 		$this->assertEquals(42, count($index->getAdded()));
 		$this->assertTrue($data->isComplete);
+
+		// Check any additional queued jobs
+		$processor->afterComplete();
+		$service = singleton('QueuedJobService');
+		$jobs = $service->getJobs();
+		$this->assertEquals(1, count($jobs));
+		$this->assertInstanceOf('SearchUpdateCommitJobProcessor', $jobs[0]['job']);
+	}
+
+	/**
+	 * Test creation of multiple commit jobs
+	 */
+	public function testMultipleCommits() {
+		$index = singleton('BatchedProcessorTest_Index');
+		$index->reset();
+
+		// Test that running a commit immediately after submitting to the indexes
+		// correctly commits
+		$first = SearchUpdateCommitJobProcessor::queue();
+		$second = SearchUpdateCommitJobProcessor::queue();
+
+		$this->assertFalse($index->getIsCommitted());
+
+		// First process will cause the commit
+		$this->assertFalse($first->jobFinished());
+		$first->process();
+		$allMessages = $first->getMessages();
+		$this->assertTrue($index->getIsCommitted());
```

Comment:
If the first commit has run, it should be committed then, not after the `$second->process()`.

When the second one is run it should see that there is in fact no more committing left to do, hence the "discarding this job" message.